galite-core/src/main/kotlin/org/kopi/galite/type/Week.kt

```diff
@@ -15,10 +15,323 @@
+
+      // 6. Find the Weekday for Y M D
+      //    H = DayOfYearNumber + (Jan1Weekday - 1)
+      //    Weekday = 1 + ((H -1) % 7)
+      val h = dayOfYearNumber + (jan1Weekday - 1)
+      val weekday = 1 + (h - 1) % 7
+
+      // 7. Find if Y M D falls in YearNumber Y-1, WeekNumber 52 or 53
+      //    if DayOfYearNumber <= (8-Jan1Weekday) and Jan1Weekday > 4
+      //       then
+      //          YearNumber = Y - 1
+      //          if Jan1Weekday = 5 or (Jan1Weekday = 6 and Y-1 is LeapYear)
+      //             then
+      //                WeekNumber = 53
+      //             else
+      //                WeekNumber = 52
+      //       else
+      //          YearNumber = Y
+      var yearNumber: Int
+      var weekNumber: Int
+      if (dayOfYearNumber <= 8 - jan1Weekday && jan1Weekday > 4) {
+        yearNumber = year - 1
+        weekNumber = if (jan1Weekday == 5 || jan1Weekday == 6 && leapYear_m_1) 53 else 52
+      } else {
+        yearNumber = year
+        weekNumber = -10000000
+      }
+
+      // 8. Find if Y M D falls in YearNumber Y+1, WeekNumber 1
+      //    if Y is LeapYear
+      //       then
+      //          I = 366
+      //       else
+      //          I = 365
+      //     if (I - DayOfYearNumber) < (4 - Weekday)
+      //        then
+      //           YearNumber = Y + 1
+      //           WeekNumber = 1
+      //        else
+      //           YearNumber = Y
+      val i = if (leapYear) 366 else 365
+      if (i - dayOfYearNumber < 4 - weekday) {
+        yearNumber = year + 1
+        weekNumber = 1
+      } /*else {
+      yearNumber = year;
+    }*/
+
+      // 9. Find if Y M D falls in YearNumber Y, WeekNumber 1 through 53
+      //    if YearNumber = Y
+      //       then
+      //          J = DayOfYearNumber + (7 - Weekday) + (Jan1Weekday -1)
+      //          WeekNumber = J / 7
+      //          if Jan1Weekday > 4
+      //                WeekNumber -= 1
+      if (yearNumber == year) {
+        val j = dayOfYearNumber + (7 - weekday) + (jan1Weekday - 1)
+        weekNumber = j / 7
+        if (jan1Weekday > 4) {
+          weekNumber -= 1
+        }
+      }
+
+      // 10. Output ISO Week Date:
+      //    if WeekNumber < 10
+      //       then
+      //          WeekNumber = "0" & WeekNumber  (WeekNumber requires 2 digits)
+      //    YearNumber - WeekNumber - Weekday    (Optional: "W" & WeekNumber)
+      return yearNumber * 53 + weekNumber - 1
+    }
+
+    /**
+     * Returns true iff the specified year is a leap year.
+     */
+    private fun isLeapYear(year: Int): Boolean {
+      return year % 4 == 0 && year % 100 != 0 || year % 400 == 0
+    }
+
+    // --------------------------------------------------------------------
+    // DATA MEMBERS
+    // --------------------------------------------------------------------
+    private val DAYS_BEFORE_MONTH = intArrayOf(0, 31, 59, 90, 120, 151, 181, 212, 243, 273, 304, 334)
+    private var calendar: GregorianCalendar? = null
```

Comment:
Not nullable , change to : 
private var calendar: GregorianCalendar = GregorianCalendar()